pkg/apis/serving/k8s_validation.go

```diff
@@ -471,6 +472,7 @@ func validateReadinessProbe(p *corev1.Probe) *apis.FieldError {
 			})
 		}
 	} else {
+		v1.SetDefaults(p)
```

Comment:
If you add changes in `pkg/apis/serving/v1/revision_defaults.go` setting default values will be taken care